File: examples/scala-crud-http-server/src/main/scala/com/jinyframework/examples/scalacrud/routers/CatRouter.scala
package com.jinyframework.examples.scalacrud.routers

import com.jinyframework.core.bio.HttpRouter
import com.jinyframework.examples.scalacrud.handlers.CatHandler

object CatRouter {
  def getRouter: HttpRouter = {
    val router = new HttpRouter()
    router.get("/", CatHandler.getCats)
    router
  }
}

File: examples/crud-http-server/src/main/java/com/jinyframework/examples/crud/router/CatRouter.java
package com.jinyframework.examples.crud.router;

import com.jinyframework.core.bio.HttpRouter;
import com.jinyframework.examples.crud.handlers.CatHandler;
import lombok.val;

public class CatRouter {
    public static HttpRouter getRouter() {
        val catHandler = new CatHandler();
        val router = new HttpRouter();
        router.get("/", catHandler::getCats);
        router.post("/", catHandler::addCat);
        router.get("/template", catHandler::template);
        return router;
    }
}
